### PR TITLE
[Screen Orientation] Add orientation and associated angle test

### PR DESCRIPTION
**Pull request by Johanna-hub**
For feedback

Test to check that orientation and angle are assigned from the options in the screen orientation values table

### CODE REVIEW

File: screen-orientation/orientation-reading.html [hunk 1]
```diff
@@ -20,6 +22,41 @@
   assert_true(angle == 0 || angle == 90 || angle == 180 || angle == 270);
 }, "Test screen.orientation default values.");
 
+promise_test(async t => {
+  await test_driver.bless("request full screen", () => {
+    return document.documentElement.requestFullscreen();
+  });
+  let primaryOrientation;
```
Comment (marcoscaceres) [inline screen-orientation/orientation-reading.html]: Could maybe instead do:

```JS
  await screen.orientation.lock("portrait-primary");
  const primaryOrientation = screen.orientation.angle === 0 
    ? "portrait-primary"
    : "landscape-primary";
```

But I'm a bit confused... if you do:

```
await screen.orientation.lock("portrait-primary");
```

Won't that assure the orientation is "portrait-primary"? 

And even if not assured, it's probably better to the value straight from "the horse's mouth"... i.e., `primaryOrientation` should be `screen.orientation.type`, right? Or am I missing something?

File: screen-orientation/orientation-reading.html [hunk 2]
```diff
@@ -20,6 +22,41 @@
...
+  await screen.orientation.lock("portrait-primary");
+  if (screen.orientation.angle === 0) {
+    primaryOrientation = "portrait-primary";
+  } else {
+    primaryOrientation = "landscape-primary";
+  }
+  if (primaryOrientation === "portrait-primary") {
```
Comment (marcoscaceres) [inline screen-orientation/orientation-reading.html]: Note the code repetition here... and below from line 46... if we look at these two, what we actually are testing is "expected" angles.  

I would test this incrementally and get rid of the if statements - you can then split this into 4 tests. 

```JS
await screen.orientation.lock("portrait-primary");
const {angle, type} = screen.orientation;
assert_equals(type, "portrait-primary");
assert_true(angle === 90 || angle === 270);
```

And so on...

**Review by marcoscaceres — CHANGES_REQUESTED** [review]
It's great how you've captured all the angles, but I think we simplify this a bit. The test is doing a lot of work - so if we split the test up, we can check the angles independently.

File: screen-orientation/orientation-reading.html (hunk 1 above)
Comment (Johanna-hub) [reply screen-orientation/orientation-reading.html]: I wrote this to try to find out what `primaryOrientation` is - what we call in the screen orientation values table `primary orientation 1 `- and using this work out all the other angles as stated in the table. https://w3c.github.io/screen-orientation/#reading-the-screen-orientation
If when the screen is locked to `portrait-primary` the angle is 0 - then we know that this is `primary orientation 1`, so `secondary orientation 1` would then be `portrait-secondary` at 180 degrees.  `Primary orientation 2` would be `landscape primary `at either 90 or 270 and `secondary orientation 2` would be the remaining angle allocated to `landscape-secondary.`
If the screen is locked to `portrait-primary` and the angle is not 0, (it should either be 90 or 270) then `landscape-primary` would be `primary orientation 1 `at 0.  `Landscape-secondary` would be `secondary orientation 1` at 180.  `Portrait-primary` at 90 or 270 would be `primary-orientation 2` and so on.
This enables a more thorough check of the relationship of orientation to angle.

Comment (marcoscaceres) [reply screen-orientation/orientation-reading.html]: Ok, yes, I get the logic and rationale, which is perfect. 

But I'm still a bit confused (heh, I should read the spec!) about the following... if we do:

```JS
await screen.orientation.lock("portrait-primary");
```

Won't the `screen.orientation.type` be "portrait-primary"? 

If yes, then it follows that `primaryOrientation` will just be  "portrait-primary".

Comment (Johanna-hub) [reply screen-orientation/orientation-reading.html]: Yes, but we're testing for the angle not the type to decide primaryOrientation.  So once we are locked to portrait-primary - if the angle is 0 - then the primary orientation is "portrait-primary".

If the angle is 90 or 270, we know that the primary orientation is "landscape-primary" and portrait-primary is a rotation from that.

Am I confused though??

File: screen-orientation/orientation-reading.html [hunk 3]
```diff
@@ -26,36 +26,39 @@
...
+  await screen.orientation.lock(orientations.primaryOrientation2);
+  assert_true(
+    screen.orientation.angle == 90 || screen.orientation.angle == 270,
+    "Primary orientation 2 angle should be either 90 or 270"
+  );
+  const primaryOrientation2Angle = screen.orientation.angle;
+  const secondaryOrientation2Angle = primaryOrientation2Angle === 90 ? 270 : 90;
+  await screen.orientation.lock(orientations.secondaryOrientation2);
+  assert_true(
+    screen.orientation.angle == secondaryOrientation2Angle,
```
Comment (marcoscaceres) [inline screen-orientation/orientation-reading.html]: assert_equals here

File: screen-orientation/orientation-reading.html [hunk 4]
```diff
@@ -20,6 +22,44 @@
...
+          primaryOrientation2: "landscape-primary",
+          secondaryOrientation2: "landscape-secondary"
+        }
+      : {
+          secondaryOrientation1: "landscape-secondary",
+          primaryOrientation2: "portrait-primary",
+          secondaryOrientation2: "portrait-secondary"
+        };
+  await screen.orientation.lock(orientations.secondaryOrientation1);
+  assert_true(
```
Comment (marcoscaceres) [inline screen-orientation/orientation-reading.html]: assert_equals here too

**Review by marcoscaceres — CHANGES_REQUESTED** [review]
Looking really good. Two little things to fix.

File: screen-orientation/orientation-reading.html [hunk 5]
```diff
@@ -20,6 +22,44 @@
...
+    screen.orientation.angle == 180,
+    "Secondary orientation 1 angle should be 180"
+  );
+  await screen.orientation.lock(orientations.primaryOrientation2);
+  assert_true(
+    screen.orientation.angle == 90 || screen.orientation.angle == 270,
+    "Primary orientation 2 angle should be either 90 or 270"
```
Comment (marcoscaceres) [inline screen-orientation/orientation-reading.html]:
```suggestion
    "Primary orientation 2 angle must be either 90 or 270"
```

Comment (marcoscaceres) [reply screen-orientation/orientation-reading.html]: The other “should” should be “must”, per the spec.

File: screen-orientation/orientation-reading.html [hunk 6]
```diff
@@ -40,21 +40,21 @@
           secondaryOrientation2: "portrait-secondary"
         };
   await screen.orientation.lock(orientations.secondaryOrientation1);
-  assert_true(
+  assert_equals(
```
Comment (marcoscaceres) [inline screen-orientation/orientation-reading.html]: this should be:
```
assert_equals(screen.orientation.angle, 180, "Secondary orientation 1 angle must be 180") 
```
i.e., assert_equals takes 3 arguments.

**Review by marcoscaceres — COMMENTED** [review]
assert_equals need updating :)

File: screen-orientation/orientation-reading.html (hunk 6 above)
Comment (Johanna-hub) [reply screen-orientation/orientation-reading.html]: Sorry!! Wasn't thinking properly!

Comment (marcoscaceres) [reply screen-orientation/orientation-reading.html]: No problem. What reviews are for 🥳

File: screen-orientation/orientation-reading.html [hunk 7]
```diff
@@ -53,7 +54,8 @@
   const secondaryOrientation2Angle = primaryOrientation2Angle === 90 ? 270 : 90;
   await screen.orientation.lock(orientations.secondaryOrientation2);
   assert_equals(
-    screen.orientation.angle == secondaryOrientation2Angle,
+    screen.orientation.angle,
+    secondaryOrientation2Angle,
     "Secondary orientation 2 angle must be the one of 90 or 270 not set to primary orientation 2"
   );
```
Comment (marcoscaceres) [inline screen-orientation/orientation-reading.html]: The description here doesn’t seem to match the test very well... something doesn’t read well about it. Could you please rephrase it?

**Review by marcoscaceres — CHANGES_REQUESTED** [review]
One little thing

File: screen-orientation/orientation-reading.html (hunk 7 above)
Comment (Johanna-hub) [reply screen-orientation/orientation-reading.html]: I've given it a go, see what you think!

Comment (marcoscaceres) [reply screen-orientation/orientation-reading.html]: Ah! much better!

**Review by marcoscaceres — APPROVED** [review]

**Review by marcoscaceres — APPROVED** [review]

**Pull request merged**